components/explore-the-data-page/DataDownloadButton.js

```diff
@@ -1,33 +1,31 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import Papa from 'papaparse';
 
 class DataDownloadButton extends React.Component {
-  csvRows() {
-    const { data } = this.props;
-    const { records } = data;
-    const headers = Object.keys(records);
-    const columns = Object.values(records);
-    const bodyRows = columns[0].map((cell, rowIndex) => columns.map(column => `"${column[rowIndex]}"`));
-
-    return [headers].concat(bodyRows);
-  }
-
   csvContent() {
-    const csvBody = this.csvRows()
-      .map(row => row.join(','))
-      .join('\n');
-
-    return `data:text/csv;charset=utf-8,${csvBody}`;
+    // https://github.com/mholt/PapaParse/issues/175#issuecomment-75597039
+    const { data } = this.props;
+    const blob = new Blob([Papa.unparse(data)]);
+    return window.URL.createObjectURL(blob, { type: 'text/plain' });
   }
 
   render() {
-    const { fileName } = this.props;
+    const { fileName, data } = this.props;
+
+    if (!data) {
+      return (
+        <Button type="button" className="btn btn--primary btn--chart-toggle" disabled>
+          Download (CSV)
+        </Button>
+      );
+    }
```

Comment:
Since the `compressed` data is smaller than the `full` data and likely to be downloaded first, we'll allow the user to see graphs and filter data before the `full` data is downloaded, but we'll disable the download button. Once the `full` data is downloaded, we'll enable the button.